assets/app/view/axis.rb

```diff
@@ -41,45 +41,66 @@ def col_labels
       hex_x, hex_y = hex_size
 
       labels = @cols.map do |c|
-        x = (hex_x / 2) + (hex_x * (c.to_i - 1))
+        x = hex_x * (c.to_i - 1)
+        y = 0
         label = ('A'..'Z').to_a[c - 1]
-        h(:text, { attrs: { x: x, y: @font_size } }, label)
+        h(:text, { attrs: { x: x, y: y } }, label)
       end
 
-      t_x = 50 + @gap
-      t_y = (hex_y * (@rows.size.to_i + 1)) + (2 * @gap)
+      t_x = X_OFFSET + @gap
+
+      rows_offset = @layout == :flat ? (@rows.size + 1) : @rows.size
+      bottom_t_y = (hex_y * rows_offset) + (@font_size * 2) + (@gap * 2)
+
+      bottom_t_y += @font_size * 2 if @layout == :pointy
 
-      [
-        h(:g, { attrs: { transform: "translate(#{t_x} 0)" } }, labels),
-        h(:g, { attrs: { 'dominant-baseline': 'hanging', transform: "translate(#{t_x} #{t_y})" } }, labels),
-      ]
+      h(:g,
+        { attrs: { transform: "translate(#{t_x + @font_size} 0)" } },
+        [
+          h(:g,
+            { attrs: { 'dominant-baseline': 'hanging' } },
+            labels),
+          h(:g,
+            { attrs: { transform: "translate(0 #{bottom_t_y})",
+                       'dominant-baseline': 'baseline' } },
+            labels),
+        ])
     end
 
     def row_labels
       hex_x, hex_y = hex_size
 
       labels = @rows.map do |row|
-        y = (hex_y * row.to_i)
-        h(:text, { attrs: { x: @font_size, y: y } }, row)
+        x = 0
+
+        multiplier = row.to_i
+        multiplier -= (1 / 2) if @layout == :pointy
```

Comment:
you mean 0.5 ? 

1 / 2 evaluates to 0

Reply:
or do you mean multiplier * 0.5 ?

Reply:
I did mean 0.5, got lucky with JS not evaluating it to 0 where Ruby would